core/src/main/scala/scalaz/Cofoldable.scala

```diff
@@ -0,0 +1,55 @@
+package scalaz
+
+trait Cofoldable[F[_], A] extends Cofoldable1[F, A] {
+  
+  def build[B](b: B)(f: B => Option[(A, B)]): F[A]
```

Comment:
I think it should be 
```scala
def build(init: S)(step: S => Maybe[(S, A)]): F[A]
```
to be coherent with [`CorecursiveList`](https://github.com/scalaz/scalaz/blob/series/7.3.x/core/src/main/scala/scalaz/CorecursiveList.scala#L39) (that would be the canonical instance of `Cofoldable`)
(Alternatively CorecursiveList would need to be modified to match this method)

Reply:
Good idea, I can change that.